src/lib.rs

```diff
@@ -571,7 +691,7 @@ mod tests {
       ]);
 
       let c3 = store.cons(3, Index(0, 0, 0)).unwrap();
-      assert_matches!(Some(Index(0, 1, 1)), c3);
+      assert_matches!(c3, Index(0, 1, 1));
```

Comment:
I was wondering about these! But the tests seemed to pass. I guess because a variable name is a pattern that matches any value, oops 😅

Reply:
Yup, not an uncommon rust gotcha.

Reply:
I imagine it comes up less it `match` clauses
and if it does that clippy will warn for unintended variable shadowing
You can do like `matches!(Index(0,1,1), x if x == c3)` right? (Modulo availability of `==`)

Reply:
I suppose that's just an `assert_eq!` tho :P